tools/summarize_results.py

```diff
@@ -0,0 +1,149 @@
+"""Provides summaries of the results of an analysis
+
+Takes an AnalysisResultProto in either text or binary form and can
+Summarize the data in a few different ways. See USAGE string below.
+"""
+
+import sys
+
+from absl import app
+from absl import flags
+from analysis import result_pb2
+from google.protobuf import text_format
+
+USAGE = """Usage:
+
+summarize_results [--input_file=<path to input>] <mode>
+
+Available modes:
+  cost_summary - print out the total cost for each method and network model.
+  request_size_distribution <method> - print out the request size distribution for a particular method.
+  response_size_distribution <method> - print out the request size distribution for a particular method.
+  latency_distriubtion <method> <network> - print out the latency distribution for a particular method and network model.
+  cost_distriubtion <method> <network> - print out the cost distribution for a particular method and network model.
+"""
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "input_file", "-",
+    "Path to a file containing an analysis result proto or '-' to read from stdin."
+)
+
+flags.DEFINE_bool("binary", False,
+                  "If true, will parse the input file as a binary proto.")
+
+
+class NetworkResultNotFound(Exception):
+  """Could not find the specified network result."""
+
+
+class MethodResultNotFound(Exception):
+  """Could not find the specified method result."""
+
+
+def main(argv):  # pylint: disable=missing-function-docstring
+  result_proto = result_pb2.AnalysisResultProto()
+  if not FLAGS.binary:
+    input_file_contents = read_input(FLAGS.input_file)
+    text_format.Merge(input_file_contents, result_proto)
+  else:
+    input_file_contents = read_binary_input(FLAGS.input_file)
+    result_proto.ParseFromString(input_file_contents)
+
+  if len(argv) < 2:
+    return print_usage()
+
+  mode = argv[1]
```

Comment:
use a flag?

Reply:
Prefer to keep the mode stuff just as argv and it makes the command line less verbose.